### PR TITLE
Bitrunning content patch: Antag rework (READY)

### LINKED ISSUE #79465: Bitrunner Console Randomise button doesn't actually choose a random sim
Reporting client version: 514.1589

## Round ID:
[217948](https://scrubby.melonmesa.com/round/217948)

## Testmerges:
- [Changes how halloween lighting works slightly](https://www.github.com/tgstation/tgstation/pull/79340)
- [Makes lighting objects objects again](https://www.github.com/tgstation/tgstation/pull/78857)
- [Extends Halloween](https://www.github.com/tgstation/tgstation/pull/79316)

## Reproduction:
I hit random five sims in a row and got Ashen Inferno all five times - This is a 1 in 1889568 chance if you have all maps available. Prior rounds have always chosen AI if you have at least 2 points for a map, Clown Planet if you only have one point, and Gondola Asteroid if you have no points.

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Larger patch for bitrunning that addresses a few issues.

- Two new antagonists: cyber tac and netguardian
- Quantum server emag opportunity
- Modular mob packs: Like random spawners, but for groups
- Antag spawning fixed: vdom antags now have up to a 10% chance to spawn based on domains loaded
- Virtual domains are no longer all fullbright by default, only the outdoorsy ones
- Actually deletes legion map file, since it was removed in #79424

<details>
<summary>images</summary>

The netguardian prime
dreamseeker_eKi7Mhv45s

robot_48_animation

The glitch effect - this mob is being mutated
dreamseeker_NE4j4rCoez

Cyber tac (t2 antagonist)
Screenshot 2023-10-27 211732

</details>

## Why It's Good For The Game
- Bitrunning antagonists are so incredibly rare that it's underwhelming to play as one for the solid second they offer if you even get the role
- Bitrunners had basically no traitor route to follow, they became assistants with black outfits

Fixes #79465

<details>
<summary>More info</summary>

Bitrunners don't have any type of traitor options. If they're made into traitors, there's nothing bitrunner related they can do, and their access is particularly bad so it's like they're a worse assistant. I've coupled this with the bitrunning antagonist system, which is now fixed.\. Bitrunners can now attempt to coax these entities to come onto the station, however they are not given any form of allegiance for doing so (and are quite counterable).

Previously, vdom antagonists relied on so many factors to spawn that it basically wouldn't happen. Now, it runs on the server each time there is a map loaded, with increasing probability as the round progresses. This builds up the list of spawnable antagonists, of which two are new, including an entirely new giant mech megafauna. This is the first "megafauna-esque" basic mob in the game. Its AI is bad, it's really only meant to be player controlled, but you can spawn them. Being mech, they are very counterable with ion rifles and the like.

Several refactors, rewrites, and overall bug fixes are included in this PR. For instance: Legion could not spawn a crate because it would look for all living legions in the mob list. Considering few even bother with this domain, it's been removed. 

Lastly, I added a framework for making bitrunner maps more random, the modular mob spawning system, which works in conjunction with random crate locations.
</details>

## Changelog
:cl:
add: Bitrunning Patch 1 features a host of changes!
add: Added randomized mobs to virtual domains, which will be indicated with a unique icon.
add: New emag interaction with the quantum server. Antags will spawn more frequently, and they can hack themselves onto the station. You have been warned.
add: Both living and dead players can now see which mob is going to spawn an antagonist in the vdom.
add: Two new vdom antagonists: Cyber Tac and the NetGuardian. These unlock at specific thresholds.
balance: You can no longer stack copies of the same ability with bitrunning disks.
balance: Some of the disk items have been replaced with stronger versions.
fix: You can no longer spy on crew using the advanced camera console on syndicate assault.
fix: Fixed the spawning mechanism of virtual domain antagonists. You should now have a chance of playing as one. This chance increases as more domains are completed.
fix: Vdom antagonists shouldn't spawn at the end of the run any longer.
fix: The preference for vdom antagonists has been changed to factor in the new types. Check your preferences!
fix: The quantum server will now show its balloon alerts to all observers.
fix: Random domains should be fully random again.
/:cl:

### CODE REVIEW

**Comment by Thunder12345** [comment]
Why are you writing an entire duplicate modular map loading system instead of using the one that already exists

**Comment by Loafin34** [comment]
i mean, the easiest traitor thing to do is either A. syndicate-backed challenge for gamer loot to fight people with, B. a chip to put in the server to make the video game into real life, or C. trap someone within the matrix using a special gun where they fight their way out to live, like heretic sacc'ing.

**Comment by Stalkeros2** [comment]
> balance: You can no longer stack abilities with bitrunning disks.

That was not a feature??????

**Comment by Timberpoes** [comment]
Crashes the MC which kills servers at current commit.

https://tgstation13.org/raw-logs/sybil/data/logs/2023/10/27/round-217621/dd.log
https://tgstation13.org/raw-logs/sybil/data/logs/2023/10/27/round-217626/dd.log

```
Infinite loop suspected--switching proc to background.
If it is not an infinite loop, either do 'set background=1' or set world.loop_checks=0.
proc name: spawn mobs (/datum/modular_mob_segment/proc/spawn_mobs)
  source file: code/modules/bitrunning/virtual_domain/modular_mob_segment.dm,46
  usr: Alannah Dracula (/mob/living/carbon/human)
  src: /datum/modular_mob_segment/syn... (/datum/modular_mob_segment/syndicate_team)
  usr.loc: the floor (85,109,4) (/turf/open/floor/iron/dark/textured_large)
  call stack:
/datum/modular_mob_segment/syn... (/datum/modular_mob_segment/syndicate_team): spawn mobs(the shuttle floor (31,68,6) (/turf/open/floor/mineral/plastitanium))
the quantum server (/obj/machinery/quantum_server): load mob segments()
the quantum server (/obj/machinery/quantum_server): cold boot map("syndicate_assault")
the quantum console (/obj/machinery/computer/quantum_console): ui act("set_domain", /list (/list), /datum/tgui (/datum/tgui), /datum/ui_state/default (/datum/ui_state/default))
/datum/tgui (/datum/tgui): on act message("set_domain", /list (/list), /datum/ui_state/default (/datum/ui_state/default))
/datum/callback/verb_callback (/datum/callback/verb_callback): InvokeAsync()
## DEBUG: Fri Oct 27 22:44:27 2023 MC restarted. Reports:
```

File: code/modules/bitrunning/virtual_domain/modular_mob_segment.dm [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+				else
+					path = mobs[current_index]
+					current_index++
+
+				var/mob/living/mob = new path(possible_turf)
+				nearby -= possible_turf
+				spawned = TRUE
+				spawned_mob_refs.Add(WEAKREF(mob))
+
+				break
```
Comment (Timberpoes) [inline code/modules/bitrunning/virtual_domain/modular_mob_segment.dm:63]: This is likely causing infinite loops when `for(var/turf/open/possible_turf in nearby)` never finds a viable turf.

**Comment by jlsnow301** [comment]
Atomizing this as much as I can

**Pull request closed without merging**